File: seunggon/week6/easy.js
// 콜라츠 추측 (lv. 1)

// https://school.programmers.co.kr/learn/courses/30/lessons/12943

/*
1이될때까지 반복 -> 1
num % 2 == 0 => /2
num % 2 != 0 => *3 +1

count
*/

function solution(num) {
  var answer = 0;

  // 입력이 1일 때는 아예 while을 돌지 않으므로 초깃값 0을 출력
  while (num != 1 && answer != 500) {
    if (num % 2 == 0) {
      num /= 2;
    } else {
      num = num * 3 + 1;
    }

    answer++;
    console.log(answer);
  }

  return answer != 500 ? answer : -1;
}

File: ryan/week6/easy.py
"""
1-1. 입력된 수가 짝수라면 2로 나눕니다. 
1-2. 입력된 수가 홀수라면 3을 곱하고 1을 더합니다. 
2. 결과로 나온 수에 같은 작업을 1이 될 때까지 반복합니다. 
"""

def solution(num, count=0):
    if num == 1:
        return count
    
    if count == 500:
        return -1
    
    if num % 2 == 0:
        return solution(num // 2, count + 1)
    else:
        return solution(num * 3 + 1, count + 1)

print(solution(16))